torch/distributed/rpc/api.py

```diff
@@ -364,6 +364,19 @@ def remote(to, func, args=None, kwargs=None):
         kwargs (dict): is a dictionary of keyword arguments for the ``func``
                        invocation.
 
+        timeout (float, optional): timeout in seconds for this remote call. If the
+                                   creation of this
+                                   :class:`~torch.distributed.rpc.RRef` on worker
+                                   ``to`` is not successfully processed on this
+                                   worker within this timeout, then the next time
+                                   there is an attempt to use the RRef (such as
+                                   ``to_here()``), a timeout will be raised
```

Comment:
Not requesting changes to this PR if this is not supported yet. What will happen for the following code?

```python
x = rpc.remote(to, my_func, args=(...), timeout=2)
x.to_here() # blocking here
# after 2 seconds, my_func timed out
```

Will `to_here()` raise an exception after two seconds?

Reply:
No, since currently, `to_here()` does a blocking call on the future corresponding to the `to_here` message, this would not happen. 

Currently, `to_here` would only notice `rpc.remote()` timed out if the callback corresponding to the remote call fired before the call to `to_here` and appropriately set `isTimedOut` which is checked in the beginning of `toHere`. Currently the user can pass a timeout to `to_here` and have that call time out. 

However, we should be able to support this by making `to_here()` wait on a CV notification that is notified when the future completes OR when `isTimedOut` is set. Then this should work as expected. I will look into this in a follow up diff.

Reply:
Filed https://github.com/pytorch/pytorch/issues/39411 to track